script.js

```diff
@@ -12,60 +12,169 @@ const confirmationButton = document.getElementById("confirmation-button");
+  name: "",
+  price: 0,
+  quantity: 0,
+  icon: "",
+};
+let cartClicked = false;
+
+const closeModal = () => {
+  confirmationModalOverlay.classList.add("opacity-0", "pointer-events-none");
+  confirmationModalOverlay.classList.remove("opacity-100");
+
+  darkOverlay.classList.replace("opacity-60", "opacity-0");
+};
+const toggleModalVisibility = (modal, isOpen) => {
+  if (isOpen) {
+    modal.classList.remove("scale-0", "opacity-0", "pointer-events-none");
+    modal.classList.add("scale-100", "opacity-100");
+  } else {
+    modal.classList.add("scale-0", "opacity-0", "pointer-events-none");
+    modal.classList.remove("scale-100", "opacity-100");
+  }
+};
+
+const calculateTotal = () => {
+  let total = 0;
+  cartItems.forEach((item) => {
+    total += item.price * item.quantity;
+  });
+  return total;
+};
+const updateTotalDisplay = () => {
+  const total = calculateTotal();
+  document.getElementById("total-cart-display").textContent = `$${total}`;
+};
 
 addToCartButtons.forEach((button) => {
   button.addEventListener("click", (event) => {
-    selectedPack = event.target.dataset.pack;
+    selectedPack.name = event.target.dataset.pack;
+    selectedPack.price = parseFloat(event.target.dataset.price);
+    selectedPack.icon = event.target.dataset.icon;
 
-    darkOverlay.classList.remove("opacity-0");
-    darkOverlay.classList.add("opacity-60");
+    darkOverlay.classList.replace("opacity-0", "opacity-60");
 
     confirmationModalOverlay.classList.remove(
       "opacity-0",
       "pointer-events-none"
     );
     confirmationModalOverlay.classList.add("flex", "opacity-100");
 
-    confirmationText.textContent = `Are you sure you want to add ${selectedPack} to your cart?`;
+    confirmationText.textContent = `Are you sure you want to add ${selectedPack.name} to your cart?`;
   });
 });
 
 confirmationButton.addEventListener("click", () => {
-  cartItems++;
-  cartItemsNumber.textContent = cartItems;
-
   confirmationModalOverlay.classList.add("opacity-0", "pointer-events-none");
   confirmationModalOverlay.classList.remove("opacity-100");
 
-  darkOverlay.classList.add("opacity-0");
-  darkOverlay.classList.remove("opacity-60");
+  darkOverlay.classList.replace("opacity-60", "opacity-0");
 
   cartButton.classList.add("scale-125");
 
-  setTimeout(() => {
-    cartButton.classList.remove("scale-125");
-    cartItemsContainer.classList.remove("hidden");
-    cartItemsContainer.classList.add("flex");
-  }, 500);
-});
+  const existingItem = cartItems.find(
+    (item) =>
+      item.name === selectedPack.name && item.price === selectedPack.price
+  );
 
-cancelConfirmationButton.addEventListener("click", () => {
-  confirmationModalOverlay.classList.add("opacity-0", "pointer-events-none");
-  confirmationModalOverlay.classList.remove("opacity-100");
+  if (existingItem) {
+    existingItem.quantity += 1;
+    const itemElement = document.getElementById(
+      `cart-item-${selectedPack.name.replace(/\s+/g, "-").toLowerCase()}-${
```

Comment:
instead of having all this logic, why don't you refactor the data-pack attribute to a consistent format like eg: Alpha Pack to "alpha-pack"

Reply:
I replaced the logic adding dataset.name and dataset.id instead of dataset.pack. So now we have `const itemElement = document.getElementById(``cart-item-${selectedPack.id}`);